### PR TITLE
feat: merge upstream-sync template

### PR DESCRIPTION
**Pull request by jackhodgkiss**
This pull request contains the workflow template for upstream-sync.

### CODE REVIEW

File: workflow-templates/upstream-sync.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+name: Upstream Sync
+'on':
+  schedule:
+    - cron: "15 8 * * 1"
```
Comment (markgoddard) [inline workflow-templates/upstream-sync.yml:5]: Might as well have a workflow_dispatch here.

**Review by markgoddard — CHANGES_REQUESTED** [review]

File: workflow-templates/upstream-sync.properties.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+    "name": "StackHPC Sync With OpenStack Upstream",
+    "description": "Synchronise with the upstream OpenStack repository against specified releases"
```
Comment (markgoddard) [inline workflow-templates/upstream-sync.properties.json:3]: We could have this:

```
"filePatterns": [
        "^\.gitreview$"
    ]
```

File: workflow-templates/upstream-sync.yml (hunk 1 above)
Comment (jackhodgkiss) [reply workflow-templates/upstream-sync.yml:5]: Good catch. It has now been added.

File: workflow-templates/upstream-sync.properties.json (hunk 2 above)
Comment (jackhodgkiss) [reply workflow-templates/upstream-sync.properties.json]: The filePatterns has been added. It complained about the backslash being an invalid escape character. Looking at the tox.properties.sync.json I noticed there is no backslash so I just used that instead.

Comment (markgoddard) [reply workflow-templates/upstream-sync.properties.json]: Without an escape, the regex will match agitreview, bgitreview, etc. I don't think it's an issue in practice though.

**Review by markgoddard — APPROVED** [review]

**Pull request merged**